### PR TITLE
Feature/salesforce env

### PR DESCRIPTION
**Pull request by DarkSnow**
Importing this configuration without the appropriate environment variables set will result in an invalid salesforce set up and will likely cause errors.

### CODE REVIEW

**Comment by DarkSnow** [comment]
I found that the config forms load the data from the config files, as expected, but do not show the overridden value.

So, if we set a real config key in code and export configuration, the config file will still just contain the dummy data, protecting the data in the environment variables. This actually caused me some trouble when testing because as I changed values in code, the config forms and exported files didn't change.

In practice, this allows us to export the bones of the config so we don't have to set everything, while still allowing us to override what we need to without risk.

**Comment by joesb** [comment]
Well, hmm. There seems to be something not working with this arrangement …

Despite having the env vars correctly set in my Docker PHP container, the `$config` values in `settings.php` are not overriding the imported config from files. But I don't know why.

```sh
wodby@php.container:/var/www/html $ drush cget salesforce.salesforce_auth.salesforce_auth
uuid: 7a007b58-09d4-49ea-9f1b-3a231d9ea9c1
langcode: en
status: true
dependencies:
  config:
    - key.key.salesforce_pem
  module:
    - salesforce_jwt
id: salesforce_auth
label: 'Salesforce Auth'
provider: jwt
provider_settings:
  consumer_key: dummy.key
  login_user: user@nationalleadership.gov.uk
  login_url: 'https://salesforce.com'
  encrypt_key: salesforce_pem
```

And it throws an error when viewing a profile page, of course, because it can't connect to Salesforce with that config.
Screenshot 2019-12-12 at 14 46 46

**Comment by joesb** [comment]
Aah, different result, looking better, with:

```sh
wodby@php.container:/var/www/html $ drush cget --include-overridden salesforce.salesforce_auth.salesforce_auth
uuid: 7a007b58-09d4-49ea-9f1b-3a231d9ea9c1
langcode: en
status: true
dependencies:
  config:
    - key.key.salesforce_pem
  module:
    - salesforce_jwt
id: salesforce_auth
label: 'Salesforce Auth'
provider: jwt
provider_settings:
  consumer_key: XXXXXXXXX.XXXXXXXXXXXXXXXXXX.XXXXXXXXXXX
  login_user: XXXXXXXXX@weareconvivio.com.XXXXXXXXX
  login_url: 'https://test.salesforce.com'
  encrypt_key: salesforce_pem
```

(I've anonymised the result)

**Comment by joesb** [comment]
But still getting the error on a profile view

File: web/sites/default/settings.prod.php [hunk 1]
```diff
@@ -29,3 +29,7 @@
 
 // Ensure the devel config environment is off
 $config['config_split.config_split.devel']['status'] = FALSE;
+
+if (getenv('CONNECT_DOMAIN')) {
+  $settings['trusted_host_patterns'] = [getenv('CONNECT_DOMAIN')];
```
[the remark this answers is not in the document]
Comment (DarkSnow) [reply web/sites/default/settings.prod.php]: Local testing shows that the entries in trusted domains do not need to be regexp but you don't get any pattern matching. For our use, locking down access to a specific domain name, `CONNECT_DOMAIN` can be used directly.

**Comment by joesb** [comment]
As far as I'm concerned, this is now all good and ready to go.

**Pull request merged**